lib/posix-socket/include/uk/socket_driver.h

```diff
@@ -91,15 +91,44 @@ struct posix_socket_driver {
  */
 typedef int (*posix_socket_driver_init_func_t)(struct posix_socket_driver *d);
 
+/**
+ * Close the socket.
+ *
+ * @param sock
+ *  Reference to the socket.
+ */
+typedef int (*posix_socket_close_func_t)(struct posix_socket_file *sock);
+
 
 /**
  * A structure containing the functions exported by the Unikraft socket driver
  */
 struct posix_socket_ops {
 	/* The initialization function on socket registration. */
 	posix_socket_driver_init_func_t   init;
+	/* vfscore ops */
+	posix_socket_close_func_t         close;
 };
 
+
+static inline int
+posix_socket_do_close(struct posix_socket_file *sock)
+{
+	UK_ASSERT(sock);
```

Comment:
More of a question rather than a suggestion, but is there a reason to keep these `UK_ASSERT(sock);` in the `posix_socket_do_*()` functions? This point will not be reached because of the `if (unlikely(!sock))` in the `posix_socket_*()`.

Reply:
I would keep this `UK_ASSERT`. Asserts should be placed where the checked condition has to be true. So here it is directly before dereferencing `sock`. You can argue that in this specific case the caller already ensures that `sock` is valid. But I would respond that checking here does not hurt and is good design. The function works on its own and it is safe to call it from other functions without having to worry that necessary conditions are not properly checked for.